File: URI/1094/1094.c
#include<stdio.h>
#include<stdlib.h>

int contador(int numRepeticao){

    int i , quantia , totalColeho = 0 , totalRatos = 0 , totalSapos = 0 , total = 0 ;
    char animal;
    float percentC , percentR , percentS ;
    
    for(i = 0 ; i < numRepeticao ; i++){
        scanf("%d %c", &quantia , &animal);

        if(quantia >= 1 && quantia <= 15){

            if(animal == 'C'){
                totalColeho = totalColeho + quantia;
            }
            else if(animal == 'R'){
                totalRatos = totalRatos + quantia;
            }
            else if(animal == 'S'){
                totalSapos = totalSapos + quantia;
            }
            else{
                //não está no escopo de testes
            }           
        }
        else{
            return -1;
        }
    }// FIM_FOR

    // contagem total de cobaias
    total = totalColeho + totalRatos + totalSapos;
    //calculando porcentagens 
    percentC = ((totalColeho * 100.0) / total);
    percentR = ((totalRatos * 100.0) / total);
    percentS = ((totalSapos * 100.0) / total);
    
    // Printando resultados finais
    printf("Total: %d cobaias\n", total);
    printf("Total de coelhos: %d\n", totalColeho);
    printf("Total de ratos: %d\n", totalRatos);
    printf("Total de sapos: %d\n", totalSapos);
    printf("Percentual de coelhos: %.2f %%\n", percentC);
    printf("Percentual de ratos: %.2f %%\n", percentR);
    printf("Percentual de sapos: %.2f %%\n", percentS);

}
int main(){
    // num de repetições
    int n;
    scanf("%d", &n);
    contador(n);
    return 0;
}

File: URI/1094/1094.py
num_rep = int(input())
totalColeho = 0
totalSapos = 0
totalRatos = 0
for i in range(num_rep):
    
    quantia , animal = input().split()
    str(animal)

    if (animal == 'C'):
        totalColeho = totalColeho + int(quantia)
    
    elif (animal == 'S') :
        totalSapos = totalSapos + int(quantia)

    elif (animal == 'R') :
        totalRatos = totalRatos + int(quantia)



total = totalColeho + totalRatos + totalSapos
percentC = float((totalColeho * 100.0) / int(total))
percentR = float((totalRatos * 100.0) / total)
percentS = float((totalSapos * 100.0) / total)

print("Total: %d cobaias" % total)
print("Total de coelhos: %d" % totalColeho)
print("Total de ratos: %d" % totalRatos)
print("Total de sapos: %d" % totalSapos)
print("Percentual de coelhos: %.2f %%" % percentC)
print("Percentual de ratos: %.2f %%" % percentR)
print("Percentual de sapos: %.2f %%" % percentS)